### PR TITLE
Feat: Deque

### PR DESCRIPTION
**Pull request by wesleimp**
Fixes: #17

### CODE REVIEW

**Comment by wesleimp** [comment]
> The implementation looks good to me, but...
> 
> * change `delete` to `remove`
> * change `insert` to `add`
> * change `get` to `peek`
> * add the `clear` method
> 
> What do you think of those changes?

It looks good to me. I was just following the reference linked in the issue, but I'll change it in the next commit.

File: src/index.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ export * from './use-set'
 export * from './use-stack'
 export * from './use-weak-map'
 export * from './use-weak-set'
+export * from './use-deque'
```
Comment (zaguiini) [inline src/index.ts]: Please put it on alphabetical order

File: src/use-deque.ts [hunk 2]
```diff
@@ -1,22 +1,22 @@
 import { useState, useMemo } from 'react'
 import produce from 'immer'
 
-export const useDeque = <T>(initialState: T[] = []) => {
```
Comment (zaguiini) [inline src/use-deque.ts]: I don't get it... Why changing from generic to fixed type?

Comment (wesleimp) [reply src/use-deque.ts]: Drugs! I don't know why I did it

**Pull request merged**